### ISSUE TITLE
shadowsExample not working on Windows

### ISSUE THREAD

**Issue by c-mendoza** [openframeworks/openFrameworks#7151]
Very excited to try out the new ofShadows, but unfortunately the example does not seem to be working on my setup. The scene renders and has lighting applied, but there are no shadows. 

I'm on Windows 10 21H2, RTX 3070, latest drivers. 

Screenshot:

![Screenshot 2022-11-03 163122](https://user-images.githubusercontent.com/1453953/199828053-f7715270-908b-4c95-a254-0f10f464ea0a.png)

Here is the verbose log in case it is useful:

```
[verbose] ofShadow :: setLightType : 2
[verbose] ofShadow :: setDepthMapResolution : incoming: : 1024 x 1024 current: 1024 x 1024 | 0
[verbose] ofShadow :: setDepthMapResolution : incoming: : 1024 x 1024 current: 1024 x 1024 | 0
[verbose] ofShadow :: setDepthMapResolution : incoming: : 1024 x 1024 current: 1024 x 1024 | 0
[verbose] ofShadow :: setDepthMapResolution : incoming: : 1024 x 1024 current: 1024 x 1024 | 0
[notice ] ofShadow :: Allocating depth map for light type:  : 0 total shadows: 1 | 0
[verbose] ofShadow :: retainFBO : for light : 0 | 0
[verbose] ofShadow :: Allocating GL_TEXTURE_CUBE_MAP_ARRAY for light type:  : 0 | 0
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 28
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 28
[verbose] ofShader: checkProgramLinkStatus(): program 28 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_GEOMETRY_SHADER_EXT shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 31
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 31
[verbose] ofShader: linkProgram(): attaching GL_GEOMETRY_SHADER_EXT shader to program 31
[verbose] ofShader: checkProgramLinkStatus(): program 31 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 35
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 35
[verbose] ofShader: checkProgramLinkStatus(): program 35 linked
[notice ] ofShadow :: Allocating depth map for light type:  : 2 total shadows: 1 | 0
[verbose] ofShadow :: retainFBO : for light : 2 | 0
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 38
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 38
[verbose] ofShader: checkProgramLinkStatus(): program 38 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_GEOMETRY_SHADER_EXT shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 41
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 41
[verbose] ofShader: linkProgram(): attaching GL_GEOMETRY_SHADER_EXT shader to program 41
[verbose] ofShader: checkProgramLinkStatus(): program 41 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 45
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 45
[verbose] ofShader: checkProgramLinkStatus(): program 45 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 48
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 48
[verbose] ofShader: checkProgramLinkStatus(): program 48 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 51
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 51
[verbose] ofShader: checkProgramLinkStatus(): program 51 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 54
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 54
[verbose] ofShader: checkProgramLinkStatus(): program 54 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 57
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 57
[verbose] ofShader: checkProgramLinkStatus(): program 57 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 60
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 60
[verbose] ofShader: checkProgramLinkStatus(): program 60 linked
[verbose] ofShader: checkAndCreateProgram(): creating GLSL program
[verbose] ofShader: setupShaderFromSource(): GL_VERTEX_SHADER shader compiled
[verbose] ofShader: setupShaderFromSource(): GL_FRAGMENT_SHADER shader compiled
[verbose] ofShader: linkProgram(): attaching GL_VERTEX_SHADER shader to program 63
[verbose] ofShader: linkProgram(): attaching GL_FRAGMENT_SHADER shader to program 63
[verbose] ofShader: checkProgramLinkStatus(): program 63 linked
[verbose] ofShadow :: retainFBO : for light : 1 | 0
[verbose] ofShadow :: retainFBO : for light : 3 | 0
```

Thanks!

**Comment by ofTheo** [comment]
Thanks for sharing this issue @c-mendoza ! 
Tagging @NickHardeman

**Comment by NickHardeman** [comment]
Thank you for sharing. If you enable Draw Frustums, do they look correct, ie like you might expect for a camera fov.

**Comment by c-mendoza** [comment]
Hi Nick. Looks like the frustums are correct:

![Screenshot 2022-11-04 114444](https://user-images.githubusercontent.com/1453953/200017607-f63c5ec4-b115-4b18-b2fd-738989327d9b.png)

Here is a RenderDoc capture, perhaps that will help:
[capture.zip](https://github.com/openframeworks/openFrameworks/files/9939990/capture.zip)

Thanks!

**Comment by NickHardeman** [comment]
Hmm, not sure what it might be. Thank you. I will look into it.

**Comment by NickHardeman** [comment]
The shadow matrix was not being set on the shader! This pull request should fix it, just tested on Windows 10. #7153

**Comment by ofTheo** [comment]
Closed by #7153 🙂
